homeassistant/components/proximity.py

```diff
@@ -31,43 +47,41 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-def setup(hass, config):  # pylint: disable=too-many-locals,too-many-statements
-    """Get the zones and offsets from configuration.yaml."""
-    ignored_zones = []
-    if 'ignored_zones' in config[DOMAIN]:
-        for variable in config[DOMAIN]['ignored_zones']:
-            ignored_zones.append(variable)
-
+def setup_proximity_component(hass, config):
+    """Set up individual proximity component."""
     # Get the devices from configuration.yaml.
-    if 'devices' not in config[DOMAIN]:
+    if 'devices' not in config:
         _LOGGER.error('devices not found in config')
         return False
 
+    ignored_zones = []
+    if 'ignored_zones' in config:
+        for variable in config['ignored_zones']:
+            ignored_zones.append(variable)
+
     proximity_devices = []
-    for variable in config[DOMAIN]['devices']:
+    for variable in config['devices']:
         proximity_devices.append(variable)
 
     # Get the direction of travel tolerance from configuration.yaml.
-    tolerance = config[DOMAIN].get('tolerance', DEFAULT_TOLERANCE)
+    tolerance = config.get('tolerance', DEFAULT_TOLERANCE)
 
     # Get the zone to monitor proximity to from configuration.yaml.
-    proximity_zone = config[DOMAIN].get('zone', DEFAULT_PROXIMITY_ZONE)
+    proximity_zone = 'zone.' \
+        + config.get('zone', DEFAULT_PROXIMITY_ZONE).lstrip('proximity.')
```

Comment:
And I know the original code doesn't have it, but string formatting is preferred.

```python
"zone.{}".format(…)
```